pkg/controller/awselb/aws_elb_controller_test.go

```diff
@@ -0,0 +1,298 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awselb
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+
+	clustopv1 "github.com/openshift/cluster-operator/pkg/apis/clusteroperator/v1alpha1"
+	clustopclientfake "github.com/openshift/cluster-operator/pkg/client/clientset_generated/clientset/fake"
+	clustopaws "github.com/openshift/cluster-operator/pkg/clusterapi/aws"
+	mockaws "github.com/openshift/cluster-operator/pkg/clusterapi/aws/mock"
+	"github.com/openshift/cluster-operator/pkg/controller"
+
+	capicommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	capiv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	capiclientfake "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/fake"
+	capiinformers "sigs.k8s.io/cluster-api/pkg/client/informers_generated/externalversions"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+	clientgotesting "k8s.io/client-go/testing"
+)
+
+func init() {
+	log.SetLevel(log.DebugLevel)
+}
+
+const (
+	testMachineName           = "testmachine"
+	testNamespace             = "test-namespace"
+	testClusterDeploymentName = "test-cluster-deployment"
+	testClusterDeploymentUUID = types.UID("test-cluster-deployment-uuid")
+	testClusterID             = "test-cluster-id"
+	testRegion                = "us-east-1"
+	testClusterVerName        = "v3-10"
+	testClusterVerNS          = "cluster-operator"
+	testImage                 = "testAMI"
+)
+
+func TestSyncMachine(t *testing.T) {
+	cases := []struct {
+		name                         string
+		generation                   int64
+		lastSuccessfulSyncGeneration int64
+		lastSuccessfulSync           time.Time
+		resyncExpected               bool
+	}{
+		{
+			name:                         "no last sync",
+			generation:                   int64(1),
```

Comment:
Remove the `int64` type conversions throughout these test cases. They are not necessary.